tasks/ceph_deploy.py

```diff
@@ -436,6 +436,140 @@ def execute_ceph_deploy(cmd):
         execute_ceph_deploy(purgedata_nodes)
 
 
+def execute_cdeploy(admin,cmd,path):
+    """Execute ceph-deploy commands """
+    """Either use git path or repo path """
+    if path is not None:
+       ec= admin.run(
+            args=[
+                 'cd',
+                 run.Raw('~/cdtest'),
+                 run.Raw(';'),
+                '{path}/ceph-deploy/ceph-deploy'.format(path=path),
+                 run.Raw(cmd),
+                ],
+               check_status=False,
+           ).exitstatus
+    else:
+      ec= admin.run(
+              args=[
+                  'cd',
+                   run.Raw('~/cdtest'),
+                   run.Raw(';'),
+                  'ceph-deploy',
+                  run.Raw(cmd),
+                ],
+              check_status=False,
+              ).exitstatus
+    if ec != 0:
+       raise RuntimeError ("failed during ceph-deploy cmd: {cmd} , ec={ec}".format(cmd=cmd,ec=ec)) 
+
+@contextlib.contextmanager
+def cli_test(ctx, config):
+    """
+     ceph-deploy cli to exercise most commonly use cli's and ensure
+     all commands works and also startup the init system.
+        
+    """
+    log.info('Ceph-deploy Test')
+    if config.get('rhbuild'):
+        path=None
+    else:
+        path = teuthology.get_testdir(ctx)  
+    mons = ctx.cluster.only(teuthology.is_type('mon'))
+    for node,role in mons.remotes.iteritems():
+        admin=node
+        admin.run( args=[ 'mkdir', '~/', 'cdtest' ],check_status=False)
+        nodename=admin.shortname
+    system_type = teuthology.get_system_type(admin)
+    if config.get('rhbuild'):
+        admin.run(args = ['sudo', 'yum', 'install', 'ceph-deploy', '-y'])
+    log.info('system type is %s', system_type)
+    osds = ctx.cluster.only(teuthology.is_type('osd'))
+    for remote,roles in osds.remotes.iteritems():
+        devs = teuthology.get_scratch_devices(remote)
+        log.info("roles %s" , roles)
+        if (len(devs) < 3):
+            log.error('Test needs minimum of 3 devices, only found %s', str(devs))
+            raise RuntimeError ( "Needs minimum of 3 devices ")
+    new_cmd= 'new ' + nodename
+    new_mon_install = 'install --mon ' + nodename
+    new_osd_install = 'install --osd ' + nodename
+    new_admin = 'install --cli ' + nodename
```

Comment:
since this is a single-node tests, adding ``--cli`` isn't necessary, as the CLI will get installed when either mon or OSD is installed (and you are doing both).